### ISSUE TITLE
Logging is inconsistent - missing PASSED OVER info if there are no new change units

### ISSUE THREAD

**Issue by jmayday** [mongock/mongock#571]
## Description

Logging is inconsistent. If there are no new change units to be applied there will be no logging for change units which were already applied. Under some circumstances, it might be problematic, because how can I be sure if app really sees a directory with all my change units? In other words - missing logging for already executed scripts (currently) means that there is nothing to execute, but at the same time it could mean there's some misconfiguration and mongock doesn't see change units. I believe logging for already executed scripts should always take place (or be configurable). 

## PRIORITY

[NORMAL/MINOR]

### Mongock
- Mongock version 5.1.2
- Modules involved: springboot, MongoDB

## Steps to Reproduce

1. Run https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-quickstart and let the change units to be executed
2. Run it again - logs **do not contain info about change units already applied** (no "PASSED OVER" log entries)
3. Add a change unit and run the app again - now, **previously applied change units are being logged** with "PASSED OVER" status. 

## Behaviour

**Expected behavior:** I would expect that logging for already applied change units happens always, no matter if there is any new change unit to be applied or not. 

**Actual behavior:** Logging with "PASSED OVER" information does not happen if there are no new change units. 

**How often the bug happens:** N/A

Log for step 2 (Steps to Reproduce):
```
2022-08-25 10:07:04.542  INFO 97520 --- [           main] io.mongock.examples.QuickStartApp        : Starting QuickStartApp using Java 17.0.4 on MacBook-Pro-Jakub.local with PID 97520 (/Users/mayday/GitHub/mongock-examples/mongodb/springboot-quickstart/target/classes started by mayday in /Users/mayday/GitHub/mongock-examples/mongodb/springboot-quickstart)
2022-08-25 10:07:04.543 DEBUG 97520 --- [           main] io.mongock.examples.QuickStartApp        : Running with Spring Boot v2.4.13, Spring v5.3.13
2022-08-25 10:07:04.544  INFO 97520 --- [           main] io.mongock.examples.QuickStartApp        : No active profile set, falling back to default profiles: default
2022-08-25 10:07:04.724  INFO 97520 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
2022-08-25 10:07:04.742  INFO 97520 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 17 ms. Found 1 MongoDB repository interfaces.
2022-08-25 10:07:05.007  INFO 97520 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2022-08-25 10:07:05.012  INFO 97520 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2022-08-25 10:07:05.012  INFO 97520 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.55]
2022-08-25 10:07:05.040  INFO 97520 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2022-08-25 10:07:05.040  INFO 97520 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 464 ms
2022-08-25 10:07:05.088  INFO 97520 --- [           main] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[127.0.0.1:27020], mode=MULTIPLE, requiredClusterType=REPLICA_SET, serverSelectionTimeout='30000 ms', requiredReplicaSetName='myReplicaSet'}
2022-08-25 10:07:05.088  INFO 97520 --- [           main] org.mongodb.driver.cluster               : Adding discovered server 127.0.0.1:27020 to client view of cluster
2022-08-25 10:07:05.120  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:1, serverValue:1079}] to 127.0.0.1:27020
2022-08-25 10:07:05.120  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:2, serverValue:1080}] to 127.0.0.1:27020
2022-08-25 10:07:05.120  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=127.0.0.1:27020, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=15565334, setName='myReplicaSet', canonicalAddress=mongo3:27020, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=null, setVersion=1, topologyVersion=TopologyVersion{processId=630690d2192cfd43572be9a7, counter=10}, lastWriteDate=Thu Aug 25 10:07:01 CEST 2022, lastUpdateTimeNanos=35964086543666}
2022-08-25 10:07:05.120  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Adding discovered server mongo2:27019 to client view of cluster
2022-08-25 10:07:05.125  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Adding discovered server mongo3:27020 to client view of cluster
2022-08-25 10:07:05.125  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Adding discovered server mongo1:27018 to client view of cluster
2022-08-25 10:07:05.125  INFO 97520 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Canonical address mongo3:27020 does not match server address.  Removing 127.0.0.1:27020 from client view of cluster
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo3:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:5, serverValue:1081}] to mongo3:27020
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo1:27018] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:8, serverValue:766}] to mongo1:27018
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo1:27018] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:7, serverValue:767}] to mongo1:27018
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo3:27020] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=mongo3:27020, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=6163167, setName='myReplicaSet', canonicalAddress=mongo3:27020, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=null, setVersion=1, topologyVersion=TopologyVersion{processId=630690d2192cfd43572be9a7, counter=10}, lastWriteDate=Thu Aug 25 10:07:01 CEST 2022, lastUpdateTimeNanos=35964100044083}
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo2:27019] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:4, serverValue:708}] to mongo2:27019
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo2:27019] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:3, serverValue:707}] to mongo2:27019
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=mongo1:27018, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=6421666, setName='myReplicaSet', canonicalAddress=mongo1:27018, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=7fffffff0000000000000005, setVersion=1, topologyVersion=TopologyVersion{processId=630690b0d57b9e648ef771a2, counter=15}, lastWriteDate=Thu Aug 25 10:07:01 CEST 2022, lastUpdateTimeNanos=35964100048791}
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo3:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:6, serverValue:1082}] to mongo3:27020
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo2:27019] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=mongo2:27019, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=6787083, setName='myReplicaSet', canonicalAddress=mongo2:27019, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=null, setVersion=1, topologyVersion=TopologyVersion{processId=630690c0c6a1ad75f103dc08, counter=7}, lastWriteDate=Thu Aug 25 10:07:01 CEST 2022, lastUpdateTimeNanos=35964100418916}
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Setting max election id to 7fffffff0000000000000005 from replica set primary mongo1:27018
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Setting max set version to 1 from replica set primary mongo1:27018
2022-08-25 10:07:05.133  INFO 97520 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Discovered replica set primary mongo1:27018
2022-08-25 10:07:05.214  INFO 97520 --- [           main] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:9, serverValue:768}] to mongo1:27018
2022-08-25 10:07:05.246  INFO 97520 --- [           main] i.m.r.core.builder.RunnerBuilderBase     : Mongock runner COMMUNITY version[5.1.2]
2022-08-25 10:07:05.247  INFO 97520 --- [           main] i.m.r.core.builder.RunnerBuilderBase     : Running Mongock with NO metadata
2022-08-25 10:07:05.269  INFO 97520 --- [           main] org.reflections.Reflections              : Reflections took 16 ms to scan 1 urls, producing 1 keys and 2 values
2022-08-25 10:07:05.272  INFO 97520 --- [           main] org.reflections.Reflections              : Reflections took 1 ms to scan 1 urls, producing 1 keys and 2 values
2022-08-25 10:07:05.449  INFO 97520 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2022-08-25 10:07:05.454  INFO 97520 --- [           main] io.mongock.examples.QuickStartApp        : Started QuickStartApp in 1.11 seconds (JVM running for 1.388)
2022-08-25 10:07:05.466  INFO 97520 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : Mongock skipping the data migration. All change set items are already executed or there is no change set item.
2022-08-25 10:07:05.466  INFO 97520 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock releasing the lock
2022-08-25 10:07:05.466  INFO 97520 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock releasing the lock
2022-08-25 10:07:05.479  INFO 97520 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock released the lock
2022-08-25 10:07:05.479  INFO 97520 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : Mongock has finished
```

Log for step 3 (Steps to Reproduce):
```
2022-08-25 10:08:19.013  INFO 97626 --- [           main] io.mongock.examples.QuickStartApp        : Starting QuickStartApp using Java 17.0.4 on MacBook-Pro-Jakub.local with PID 97626 (/Users/mayday/GitHub/mongock-examples/mongodb/springboot-quickstart/target/classes started by mayday in /Users/mayday/GitHub/mongock-examples/mongodb/springboot-quickstart)
2022-08-25 10:08:19.013 DEBUG 97626 --- [           main] io.mongock.examples.QuickStartApp        : Running with Spring Boot v2.4.13, Spring v5.3.13
2022-08-25 10:08:19.014  INFO 97626 --- [           main] io.mongock.examples.QuickStartApp        : No active profile set, falling back to default profiles: default
2022-08-25 10:08:19.175  INFO 97626 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
2022-08-25 10:08:19.190  INFO 97626 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 13 ms. Found 1 MongoDB repository interfaces.
2022-08-25 10:08:19.414  INFO 97626 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2022-08-25 10:08:19.418  INFO 97626 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2022-08-25 10:08:19.418  INFO 97626 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.55]
2022-08-25 10:08:19.446  INFO 97626 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2022-08-25 10:08:19.446  INFO 97626 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 410 ms
2022-08-25 10:08:19.492  INFO 97626 --- [           main] org.mongodb.driver.cluster               : Cluster created with settings {hosts=[127.0.0.1:27020], mode=MULTIPLE, requiredClusterType=REPLICA_SET, serverSelectionTimeout='30000 ms', requiredReplicaSetName='myReplicaSet'}
2022-08-25 10:08:19.492  INFO 97626 --- [           main] org.mongodb.driver.cluster               : Adding discovered server 127.0.0.1:27020 to client view of cluster
2022-08-25 10:08:19.519  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:1, serverValue:1083}] to 127.0.0.1:27020
2022-08-25 10:08:19.519  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:2, serverValue:1084}] to 127.0.0.1:27020
2022-08-25 10:08:19.519  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=127.0.0.1:27020, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=13336458, setName='myReplicaSet', canonicalAddress=mongo3:27020, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=null, setVersion=1, topologyVersion=TopologyVersion{processId=630690d2192cfd43572be9a7, counter=10}, lastWriteDate=Thu Aug 25 10:08:11 CEST 2022, lastUpdateTimeNanos=36038486227666}
2022-08-25 10:08:19.520  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Adding discovered server mongo2:27019 to client view of cluster
2022-08-25 10:08:19.521  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Adding discovered server mongo3:27020 to client view of cluster
2022-08-25 10:08:19.521  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Adding discovered server mongo1:27018 to client view of cluster
2022-08-25 10:08:19.522  INFO 97626 --- [127.0.0.1:27020] org.mongodb.driver.cluster               : Canonical address mongo3:27020 does not match server address.  Removing 127.0.0.1:27020 from client view of cluster
2022-08-25 10:08:19.541  INFO 97626 --- ['}-mongo1:27018] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:8, serverValue:769}] to mongo1:27018
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo2:27019] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:3, serverValue:710}] to mongo2:27019
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo3:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:6, serverValue:1085}] to mongo3:27020
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo2:27019] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:4, serverValue:709}] to mongo2:27019
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo2:27019] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=mongo2:27019, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=5864041, setName='myReplicaSet', canonicalAddress=mongo2:27019, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=null, setVersion=1, topologyVersion=TopologyVersion{processId=630690c0c6a1ad75f103dc08, counter=7}, lastWriteDate=Thu Aug 25 10:08:11 CEST 2022, lastUpdateTimeNanos=36038509335750}
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo1:27018] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:7, serverValue:770}] to mongo1:27018
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo3:27020] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:5, serverValue:1086}] to mongo3:27020
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=mongo1:27018, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=6714667, setName='myReplicaSet', canonicalAddress=mongo1:27018, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=7fffffff0000000000000005, setVersion=1, topologyVersion=TopologyVersion{processId=630690b0d57b9e648ef771a2, counter=15}, lastWriteDate=Thu Aug 25 10:08:11 CEST 2022, lastUpdateTimeNanos=36038509871625}
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo3:27020] org.mongodb.driver.cluster               : Monitor thread successfully connected to server with description ServerDescription{address=mongo3:27020, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=13, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=6948125, setName='myReplicaSet', canonicalAddress=mongo3:27020, hosts=[mongo2:27019, mongo3:27020, mongo1:27018], passives=[], arbiters=[], primary='mongo1:27018', tagSet=TagSet{[]}, electionId=null, setVersion=1, topologyVersion=TopologyVersion{processId=630690d2192cfd43572be9a7, counter=10}, lastWriteDate=Thu Aug 25 10:08:11 CEST 2022, lastUpdateTimeNanos=36038510077958}
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Setting max election id to 7fffffff0000000000000005 from replica set primary mongo1:27018
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Setting max set version to 1 from replica set primary mongo1:27018
2022-08-25 10:08:19.542  INFO 97626 --- ['}-mongo1:27018] org.mongodb.driver.cluster               : Discovered replica set primary mongo1:27018
2022-08-25 10:08:19.624  INFO 97626 --- [           main] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:9, serverValue:771}] to mongo1:27018
2022-08-25 10:08:19.653  INFO 97626 --- [           main] i.m.r.core.builder.RunnerBuilderBase     : Mongock runner COMMUNITY version[5.1.2]
2022-08-25 10:08:19.654  INFO 97626 --- [           main] i.m.r.core.builder.RunnerBuilderBase     : Running Mongock with NO metadata
2022-08-25 10:08:19.682  INFO 97626 --- [           main] org.reflections.Reflections              : Reflections took 21 ms to scan 1 urls, producing 1 keys and 3 values
2022-08-25 10:08:19.685  INFO 97626 --- [           main] org.reflections.Reflections              : Reflections took 1 ms to scan 1 urls, producing 1 keys and 3 values
2022-08-25 10:08:19.889  INFO 97626 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2022-08-25 10:08:19.895  INFO 97626 --- [           main] io.mongock.examples.QuickStartApp        : Started QuickStartApp in 1.035 seconds (JVM running for 1.416)
2022-08-25 10:08:19.911  INFO 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock trying to acquire the lock
2022-08-25 10:08:19.940  INFO 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock acquired the lock until: Thu Aug 25 10:09:19 CEST 2022
2022-08-25 10:08:19.940  INFO 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Starting mongock lock daemon...
2022-08-25 10:08:19.940 DEBUG 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Mongock(daemon) ensuring lock
2022-08-25 10:08:19.940 DEBUG 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Mongock trying to refresh the lock
2022-08-25 10:08:19.942  INFO 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : Mongock starting the data migration sequence id[2022-08-25T10:08:19.653419-781]...
2022-08-25 10:08:19.943  INFO 97626 --- [           main] i.m.r.c.executor.ChangeLogRuntimeImpl    : method[io.mongock.examples.changeunits.ClientInitializerChangeUnit] with arguments: []
2022-08-25 10:08:19.943  INFO 97626 --- [       Thread-1] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:10, serverValue:772}] to mongo1:27018
2022-08-25 10:08:19.943  INFO 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : PASSED OVER - {"id"="client-initializer_before", "type"="before-execution", "author"="mongock", "class"="ClientInitializerChangeUnit", "method"="beforeExecution"}
2022-08-25 10:08:19.944  INFO 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : PASSED OVER - {"id"="client-initializer", "type"="execution", "author"="mongock", "class"="ClientInitializerChangeUnit", "method"="execution"}
2022-08-25 10:08:19.944  INFO 97626 --- [           main] i.m.r.c.executor.ChangeLogRuntimeImpl    : method[io.mongock.examples.changeunits.ClientUpdaterChangeUnit] with arguments: []
2022-08-25 10:08:19.944  INFO 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : PASSED OVER - {"id"="client-updater", "type"="execution", "author"="mongock", "class"="ClientUpdaterChangeUnit", "method"="execution"}
2022-08-25 10:08:19.944  INFO 97626 --- [           main] i.m.r.c.executor.ChangeLogRuntimeImpl    : method[io.mongock.examples.changeunits.MyNewChangeUnit] with arguments: []
2022-08-25 10:08:19.944 DEBUG 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : executing changeSet[new-change-unit]
2022-08-25 10:08:19.949 DEBUG 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Mongock refreshed the lock until: Thu Aug 25 10:09:19 CEST 2022
2022-08-25 10:08:19.950 DEBUG 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Calculating millis for lock daemon to sleep
2022-08-25 10:08:19.950 DEBUG 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Difference now[Thu Aug 25 10:08:19 CEST 2022] and Lock's expiry-date[Thu Aug 25 10:09:19 CEST 2022]: 59990ms 
2022-08-25 10:08:19.950 DEBUG 97626 --- [       Thread-1] i.m.driver.core.lock.LockManagerDefault  : Mongock lock daemon sleeping for 19800ms
2022-08-25 10:08:19.965  INFO 97626 --- [           main] i.m.r.c.executor.ChangeLogRuntimeImpl    : method[execution] with arguments: [org.springframework.data.mongodb.core.MongoTemplate]
2022-08-25 10:08:19.967 DEBUG 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Ensuring the lock
2022-08-25 10:08:19.967 DEBUG 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Dont need to refresh the lock at[Thu Aug 25 10:08:19 CEST 2022], acquired until: Thu Aug 25 10:09:19 CEST 2022
2022-08-25 10:08:19.972 DEBUG 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Ensuring the lock
2022-08-25 10:08:19.972 DEBUG 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Dont need to refresh the lock at[Thu Aug 25 10:08:19 CEST 2022], acquired until: Thu Aug 25 10:09:19 CEST 2022
2022-08-25 10:08:20.049 DEBUG 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : successfully executed changeSet[new-change-unit]
2022-08-25 10:08:20.049  INFO 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : APPLIED - {"id"="new-change-unit", "type"="execution", "author"="jmayday", "class"="MyNewChangeUnit", "method"="execution"}
2022-08-25 10:08:20.064  INFO 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock releasing the lock
2022-08-25 10:08:20.064  INFO 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock releasing the lock
2022-08-25 10:08:20.070  INFO 97626 --- [           main] i.m.driver.core.lock.LockManagerDefault  : Mongock released the lock
2022-08-25 10:08:20.071  INFO 97626 --- [           main] i.m.r.c.e.o.migrate.MigrateExecutorBase  : Mongock has finished
```

**Comment by osantana85** [comment]
Hello @jmayday, thanks for raise the issue. You are right, we have a two step validation process to avoid taking the lock when it isn't necessary, and when there is no changeunit to be executed, we are not logging the already executed ones. We'll work in it and let you know when it'll be released.

**Comment by gaganis** [comment]
Hello @osantana85, this has also impacting us. We have also experieced a situation where we cannot really tell whether we have a problem or if we are just passing over all changesets. 

This would be nice to have also for us. Thanks

**Comment by gaganis** [comment]
For me also just changing the log message to either say we found none or we skipped X changesets would be enough. Even though actually printing what is passed over would be better.

**Comment by osantana85** [comment]
Hello @gaganis , we have been waiting to fix it in the next release because we have refactored the executor, so we expect to release it in the next days. I’ll let you know when it will be available.

**Comment by gaganis** [comment]
FYI in case you are interested we experienced a case where nothing was discovered when it should. 

When running our springboot app from a jar changeunits were not discovered at all when they should. They were properly discovered when running from an IDE or through maven.

We have discovered that it was a problem in the org.reflections that was updated by another dependency to `0.10.2`. Forcing version `0.10.1` fixed the problem for us.
